Test.System.Containers/ContainerHelpersTests.cs

```diff
@@ -0,0 +1,53 @@
+using System.Containers;
+using System.Text;
+
+namespace Test.System.Containers
+{
+    [TestClass]
+    public class ContainerHelpersTests
+    {
+
+        [TestMethod]
+        [DataRow("https://mcr.microsoft.com/dotnet/runtime:6.0", true, "https://mcr.microsoft.com", "dotnet/runtime", "6.0")]
+        [DataRow("https://mcr.microsoft.com/dotnet/runtime", true, "https://mcr.microsoft.com", "dotnet/runtime", "")]
+        [DataRow("https://mcr.microsoft.com/", false, "", "", "")] // no image = nothing resolves
+        [DataRow("docker://mcr.microsoft.com/dotnet/runtime", true, "docker://mcr.microsoft.com", "dotnet/runtime", "")]
+        public void TryParseFullyQualifiedContainerName(string fullyQualifiedName, bool expectedReturn, string expectedRegistry, string expectedImage, string expectedTag)
+        {
+            Assert.AreEqual(ContainerHelpers.TryParseFullyQualifiedContainerName(fullyQualifiedName, out string containerReg, out string containerName, out string containerTag), expectedReturn);
+            Assert.AreEqual(containerReg, expectedRegistry);
+            Assert.AreEqual(containerName, expectedImage);
+            Assert.AreEqual(containerTag, expectedTag);
+        }
+
+        [TestMethod]
+        [DataRow("dotnet/runtime", true)]
+        [DataRow("foo/bar", true)]
```

Comment:
Probably worth having a no-slash option here

```suggestion
        [DataRow("foo/bar", true)]
        [DataRow("registry", true)]
```